upcloud_api/api.py

```diff
@@ -10,14 +11,19 @@ class API:
     Handles basic HTTP communication with API.
     """
 
-    api_root = 'https://api.upcloud.com/1.3'
+    user_agent = f'upcloud-python-api/{__version__}'
 
-    def __init__(self, token, timeout=None):
+    def __init__(
+        self, token, timeout=None, api_root='https://api.upcloud.com/1.3', custom_user_agent=''
```

Comment:
Why not do custom_user_agent=None ?

Reply:
Only None can be None, string is an empty one by default.